app/testonly/LoginService.scala

```diff
@@ -16,71 +16,56 @@
 
 package testonly
 
-import java.util.UUID
+import java.util.UUID.randomUUID
 
 import com.google.inject.Inject
 import javax.inject.Singleton
-import org.joda.time.DateTime
+import org.joda.time.DateTime.now
 import play.api.http.HeaderNames
 import play.api.libs.json.{Json, _}
 import play.api.mvc.Session
 import uk.gov.hmrc.http.{HeaderCarrier, SessionKeys}
-import uk.gov.hmrc.play.bootstrap.http.HttpClient
 import uk.gov.hmrc.play.bootstrap.config.ServicesConfig
+import uk.gov.hmrc.play.bootstrap.http.HttpClient
 
 import scala.concurrent.{ExecutionContext, Future}
 
 @Singleton
-class LoginService @Inject() (
-    httpClient:     HttpClient,
-    servicesConfig: ServicesConfig
-)(implicit ec: ExecutionContext) {
+class LoginService @Inject()(httpClient: HttpClient, servicesConfig: ServicesConfig)(implicit ec: ExecutionContext) {
 
   def logIn(tu: TestUser): Future[Session] = for {
     (at, au, gt) <- callAuthLoginApi(tu)
   } yield buildSession(au, at, gt, tu)
 
-  private def buildSession(
-      authorityUri: AuthorityUri,
-      authToken:    AuthToken,
-      gatewayToken: GatewayToken,
-      tu:           TestUser
-  ) = {
+  private def buildSession(authorityUri: AuthorityUri, authToken: AuthToken, gatewayToken: GatewayToken, user: TestUser) =
     Session(Map(
-      SessionKeys.sessionId -> s"session-${UUID.randomUUID}",
-      SessionKeys.authProvider -> "GGW",
+      SessionKeys.sessionId -> s"session-$randomUUID",
       SessionKeys.userId -> authorityUri.v,
       SessionKeys.authToken -> authToken.v,
-      SessionKeys.lastRequestTimestamp -> DateTime.now.getMillis.toString,
-      SessionKeys.token -> gatewayToken.v,
-      SessionKeys.affinityGroup -> tu.affinityGroup.v,
-      SessionKeys.name -> tu.authorityId.v
-    ))
-  }
+      SessionKeys.lastRequestTimestamp -> now.getMillis.toString))
```

Comment:
Please inject `clock:Clock` and compute time as: `now(clock)`. It is easier to test later.